src/connectors/ledger_subprovider.js

```diff
@@ -0,0 +1,109 @@
+import { LedgerSubprovider as LedgerSubprovider0x } from '@0x/subproviders/lib/src/subproviders/ledger' // https://github.com/0xProject/0x-monorepo/issues/1400
+import { Transaction } from 'ethereumjs-tx'
+import Common from 'ethereumjs-common'
+import { addressUtils } from '@0x/utils';
+import web3 from 'web3'
+import { 
+    LedgerSubproviderErrors,
+    WalletSubproviderErrors
+} from '@0x/subproviders/lib/src/types'
+
+/**
+ * A custom Ledger subprovider, inheriting from the 0x Subprovider.
+ * 
+ * Supports chains with a chainId > 255, and mitigates some issues with the
+ * LedgerJS library.
+ */
+class LedgerSubprovider extends LedgerSubprovider0x {
+    chainId
+
+    constructor(config) {
+        super(config)
+        this.chainId = config.chainId
+    }
+
+    async signTransactionAsync(txParams) {
+        LedgerSubprovider._validateTxParams(txParams)
+        if (txParams.from === undefined || !addressUtils.isAddress(txParams.from)) {
+            throw new Error(WalletSubproviderErrors.FromAddressMissingOrInvalid);
+        }
+        const initialDerivedKeyInfo = await this._initialDerivedKeyInfoAsync();
+        const derivedKeyInfo = this._findDerivedKeyInfoForAddress(initialDerivedKeyInfo, txParams.from);
+
+        this._ledgerClientIfExists = await this._createLedgerClientAsync();
+
+        try {
+            const fullDerivationPath = derivedKeyInfo.derivationPath
+            const common = Common.forCustomChain(
```

Comment:
If #206 will be merged, let's use [this](https://github.com/keep-network/tbtc-dapp/pull/206/files#diff-126ade12ebab43a145465e531303bddaR10-R20) to create transaction object.

Reply:
Addressed in #206 with fb516a1